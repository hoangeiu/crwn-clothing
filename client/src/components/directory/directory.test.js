import React from "react";
import { shallow } from "enzyme";

import { Directory } from "./directory.component";
import MenuItem from "../menu-item/menu-item.component";

describe("render Directory component", () => {
  let wrapper;
  let mockSections;
  beforeEach(() => {
    mockSections = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];

    const mockProps = {
      sections: mockSections,
    };

    wrapper = shallow(<Directory {...mockProps} />);
  });

  it("should Directory component render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render an equal number of MenuItem component as sections prop", () => {
    expect(wrapper.find(MenuItem).length).toEqual(mockSections.length);
  });
});
