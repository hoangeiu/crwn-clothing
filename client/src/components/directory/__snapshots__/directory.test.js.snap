// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Directory component should Directory component render without error 1`] = `
<div
  className="directory-menu"
>
  <withRouter(MenuItem)
    key="1"
  />
  <withRouter(MenuItem)
    key="2"
  />
  <withRouter(MenuItem)
    key="3"
  />
  <withRouter(MenuItem)
    key="4"
  />
  <withRouter(MenuItem)
    key="5"
  />
</div>
`;
